app/javascript/components/items/Table.js

```diff
@@ -0,0 +1,184 @@
+
+  const requestSort = (key) => {
+    let direction = 'ascending';
+    if (
+      sortConfig &&
+      sortConfig.key === key &&
+      sortConfig.direction === 'ascending'
+    ) {
+      direction = 'descending';
+    }
+    setSortConfig({ key, direction });
+  };
+
+  return { items, sortedItems, requestSort, sortConfig };
+};
+
+const ItemRow = (props) => {
+  const [state, setState] = React.useState({ item: props.item });
+  const item = state.item;
+  const bought = item.bought ? "item-bought" : "";
+  const checked = item.bought ? "checked" : "";
+  const rowClasses = `items-table-item ${bought}`;
+  const checkboxId = "item_bought_" + item.id;
+
+  function toggleBought(item) {
+    const toggleItem = item;
+    toggleItem.bought = !toggleItem.bought;
+    const data = { bought: toggleItem.bought };
+    fetch("/items/" + toggleItem.id, {
+      method: "PUT",
+      body: JSON.stringify(data),
+      headers: {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "X-CSRF-Token": props.csrf,
+      }
+    }).then(response => setState({ item: toggleItem }));
+  };
+
+  return (
+    <tr className={rowClasses}>
+      <th scope="row">{props.index + 1}</th>
+      <td><EditableField id={item.id} field="name" text={item.name} csrf={props.csrf} /></td>
+      <td><EditableField id={item.id} field="person" text={item.person} csrf={props.csrf} /></td>
+      <td><EditableField id={item.id} field="department" text={item.department} csrf={props.csrf} /></td>
+      <td><input type="checkbox" value="1" checked={checked} id={checkboxId} onChange={() => toggleBought(item)} /></td>
+    </tr>
+  );
+};
+
+const Table = (props) => {
+  const csrf = props.csrf;
+  const { items, requestSort, sortConfig, setSortConfig } = useSortableData(props.items);
+  const getClassNamesFor = (name) => {
+    if (!sortConfig) {
+      return;
+    }
+    return sortConfig.key === name ? sortConfig.direction : undefined;
+  };
+  const handleSubmit = (e) => {
+    console.log(e)
+    const nameField = e.target.closest("tr").querySelector("input[name=itemNameField");
+    const personField = e.target.closest("tr").querySelector("input[name=itemPersonField");
+    const departmentField = e.target.closest("tr").querySelector("input[name=itemDepartmentField");
+    const data = {
+      list_id: props.id,
+      name: nameField.value,
+      person: e.target.closest("tr").querySelector("input[name=itemPersonField").value,
+      department: e.target.closest("tr").querySelector("input[name=itemDepartmentField").value
+    };
+    fetch("/items/", {
+      method: "POST",
+      body: JSON.stringify(data),
+      headers: {
+        "Content-Type": "application/json",
+        "Accept": "application/json",
+        "X-CSRF-Token": props.csrf,
+      }
+    })
+    .then(response => response.json())
+    .then(json_response => {
+      data.id = json_response.id;
+      items.push(data);
+      requestSort("name");
```

Comment:
This will reset the sorting to name.  If you want to make this sort on the existing key/direction, you'll need to use `sortConfig.key` instead of a hard-coded name.

Reply:
That makes sense, good catch. Fixing.